numpy/random/mtrand/distributions.c

```diff
@@ -719,26 +720,32 @@ double rk_wald(rk_state *state, double mean, double scale)
 
 long rk_zipf(rk_state *state, double a)
 {
-    double T, U, V;
-    long X;
+    double T, U, V, X;
```

Comment:
Can you move these declarations inside the loop now?

Reply:
Yes, can do. I wonder why `T` is initialized before entry. I suspect an historical artifact ...